compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java

```diff
@@ -144,16 +147,31 @@ void instrumentPackage(BIRPackage pkg) {
             rewriteAsyncInvocations(func, null, pkg);
             rewriteObservableFunctionInvocations(func, pkg);
             if (ENTRY_POINT_MAIN_METHOD_NAME.equals(func.name.value)) {
-                rewriteObservableFunctionBody(func, pkg, null, func.name.value, false, false, true, false);
+                rewriteObservableFunctionBody(func, pkg, null, func.name.value, "", false, false, true, false);
             } else if ((func.flags & Flags.WORKER) == Flags.WORKER) {   // Identifying lambdas generated for workers
-                rewriteObservableFunctionBody(func, pkg, null, func.workerName.value, false, false, false, true);
+                rewriteObservableFunctionBody(func, pkg, null, func.workerName.value, "", false, false, false, true);
             }
         }
         for (BIRTypeDefinition typeDef : pkg.typeDefs) {
             if ((typeDef.flags & Flags.CLASS) != Flags.CLASS && typeDef.type.tag == TypeTags.OBJECT) {
                 continue;
             }
             boolean isService = (typeDef.type.flags & Flags.SERVICE) == Flags.SERVICE;
+            String serviceName = null;
+            if (isService) {
+                for (BIRNode.BIRAnnotationAttachment annotationAttachment : typeDef.annotAttachments) {
+                    if (!DISPLAY_ANNOTATION.equals(annotationAttachment.annotTagRef.value)) {
+                        continue;
+                    }
+                    serviceName = ((BIRNode.BIRAnnotationLiteralValue) (((BIRNode.BIRAnnotationRecordValue)
+                            annotationAttachment.annotValues.get(0)).annotValueEntryMap).get("label")).value.toString();
+                    break;
```

Comment:
Shall we refactor this ? I think following is cleaner.
```suggestion
                    if (DISPLAY_ANNOTATION.equals(annotationAttachment.annotTagRef.value)) {
                        serviceName = ((BIRNode.BIRAnnotationLiteralValue) (((BIRNode.BIRAnnotationRecordValue)
                                annotationAttachment.annotValues.get(0)).annotValueEntryMap).get("label")).value.toString();
                        break;
                    }
```

Reply:
FIxed with 3a5fad4b8ee95545ae441daea7a28e24052367f6